pkg/gardenlet/controller/shoot/shoot_control_prepare_migration.go

```diff
@@ -0,0 +1,393 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shoot
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardencorev1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	"github.com/gardener/gardener/pkg/operation"
+	botanistpkg "github.com/gardener/gardener/pkg/operation/botanist"
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/utils"
+	utilerrors "github.com/gardener/gardener/pkg/utils/errors"
+	"github.com/gardener/gardener/pkg/utils/flow"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	retryutils "github.com/gardener/gardener/pkg/utils/retry"
+	"github.com/gardener/gardener/pkg/version"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (c *Controller) prepareShootForMigration(shoot *gardencorev1beta1.Shoot, logger *logrus.Entry) (reconcile.Result, error) {
+
+	var (
+		respectSyncPeriodOverwrite = c.respectSyncPeriodOverwrite()
+		failed                     = common.IsShootFailed(shoot)
+		ignored                    = common.ShouldIgnoreShoot(respectSyncPeriodOverwrite, shoot)
+	)
+
+	if failed || ignored {
+		return reconcile.Result{}, fmt.Errorf("shoot %s is failed or ignored, will skip migration preparation", shoot.GetName())
+	}
+
+	targetSeedAvailable := c.isSeedAvailable(shoot.Spec.SeedName, logger)
+	if !targetSeedAvailable {
+		return reconcile.Result{}, fmt.Errorf("target Seed is not available to host the Shoot's %s Control Plane", shoot.GetName())
+	}
+
+	o, err := operation.New(shoot, shoot.Status.SeedName, c.config, logger, c.k8sGardenClient, c.k8sGardenCoreInformers.Core().V1beta1(), c.identity, c.secrets, c.imageVector)
+	if err != nil {
+		return reconcile.Result{}, utilerrors.WithSuppressed(err, c.updateShootStatusError(shoot, fmt.Sprintf("Could not initialize a new operation for preparation of Shoot control plane migration: %s", err.Error())))
```

Comment:
```suggestion
		return reconcile.Result{}, utilerrors.WithSuppressed(err, c.updateShootStatusError(shoot, fmt.Sprintf("Could not initialize a new operation for preparation of Shoot Control Plane migration: %s", err.Error())))
```
nit: Let's follow the same pattern with naming the `Control Plane` in logs/error (either capital or lower case letters)